### PR TITLE
Interprocess communication sync

### PR DESCRIPTION
**Pull request by satyasivaprasad**

### CODE REVIEW

File: .gradle/7.6/checksums/checksums.lock
Comment (shivakrishnaah) [inline .gradle/7.6/checksums/checksums.lock:1]: add these to gitignore

Comment (satyasivaprasad) [reply .gradle/7.6/checksums/checksums.lock:1]: done

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,37 @@
+HELP.md
+.gradle
```
Comment (shivakrishnaah) [inline .gitignore:2]: add .gradle/ as well

**Review by shivakrishnaah — CHANGES_REQUESTED** [review]

File: inventory-service/src/main/java/ms/learnings/inventoryservice/controller/InventoryController.java [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
     private final InventoryService inventoryService;
 
-    @GetMapping("/{sku-code}")
+    // http://localhost:8082/api/inventory/iphone-13,iphone13-red
+
+    // http://localhost:8082/api/inventory?skuCode=iphone-13&skuCode=iphone13-red
+    @GetMapping
     @ResponseStatus(HttpStatus.OK)
-    public boolean isInStock(@PathVariable("sku-code") String skuCode) {
+    public List<InventoryResponse> isInStock(@RequestParam List<String> skuCode) {
```
Comment (shivakrishnaah) [inline inventory-service/src/main/java/ms/learnings/inventoryservice/controller/InventoryController.java]: can skuCode can be duplicated? if not probably convert it into a Set

File: inventory-service/src/main/java/ms/learnings/inventoryservice/dto/InventoryResponse.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class InventoryResponse {
```
Comment (shivakrishnaah) [inline inventory-service/src/main/java/ms/learnings/inventoryservice/dto/InventoryResponse.java:12]: We want this to be unique based on skuCode

File: order-service/build/classes/java/main/ms/learnings/orderservice/OrderServiceApplication.class
Comment (shivakrishnaah) [inline order-service/build/classes/java/main/ms/learnings/orderservice/OrderServiceApplication.class:1]: why do we have class files here

File: order-service/src/main/java/ms/learnings/orderservice/client/InventoryClient.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import ms.learnings.orderservice.config.ClientConfiguration;
+import ms.learnings.orderservice.dto.InventoryResponse;
+import org.springframework.cloud.openfeign.CollectionFormat;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@FeignClient(value = "inventory-service", url = "http://localhost:8082/api/")
```
Comment (shivakrishnaah) [inline order-service/src/main/java/ms/learnings/orderservice/client/InventoryClient.java]: pull the url from properties files

File: order-service/src/main/java/ms/learnings/orderservice/config/ClientConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package ms.learnings.orderservice.config;
+
+import feign.Contract;
+import org.springframework.context.annotation.Bean;
+
+public class ClientConfiguration {
```
Comment (shivakrishnaah) [inline order-service/src/main/java/ms/learnings/orderservice/config/ClientConfiguration.java]: Where are you using ?

File: .gitignore (hunk 1 above)
Comment (satyasivaprasad) [reply .gitignore:2]: Done

File: inventory-service/src/main/java/ms/learnings/inventoryservice/controller/InventoryController.java (hunk 2 above)
Comment (satyasivaprasad) [reply inventory-service/src/main/java/ms/learnings/inventoryservice/controller/InventoryController.java]: Changed to Set

File: order-service/src/main/java/ms/learnings/orderservice/client/InventoryClient.java (hunk 4 above)
Comment (satyasivaprasad) [reply order-service/src/main/java/ms/learnings/orderservice/client/InventoryClient.java]: Done

File: order-service/src/main/java/ms/learnings/orderservice/config/ClientConfiguration.java (hunk 5 above)
Comment (satyasivaprasad) [reply order-service/src/main/java/ms/learnings/orderservice/config/ClientConfiguration.java]: Removed, not needed

File: inventory-service/src/main/java/ms/learnings/inventoryservice/dto/InventoryResponse.java (hunk 3 above)
Comment (satyasivaprasad) [reply inventory-service/src/main/java/ms/learnings/inventoryservice/dto/InventoryResponse.java:12]: Done

File: order-service/build/classes/java/main/ms/learnings/orderservice/OrderServiceApplication.class
Comment (satyasivaprasad) [reply order-service/build/classes/java/main/ms/learnings/orderservice/OrderServiceApplication.class:1]: added to gitignore

**Review by shivakrishnaah — APPROVED** [review]

**Pull request merged**